python-package/lightgbm/callback.py

```diff
@@ -2,14 +2,27 @@
 """Callbacks library."""
 import collections
 from operator import gt, lt
+from typing import Any, Callable, List, Union
 
 from .basic import _ConfigAliases, _log_info, _log_warning
 
 
+class CallbackWithAttributes():
+    """Allows the creation of function attributes without making mypy raise errors."""
```

Comment:
```suggestion
    """Bare type used to tell type checkers which attributes callback functions have"""
```

Similar suggestion here. What do you think about this proposed description? I think that "prevents mypy errors" is too limited of an explanation and doesn't explain exactly what's happening. The immediate reason to add these changes is to address warnings found by `mypy`, but `mypy` is not the only thing in the Python ecosystem that takes advantage of libraries' type hints if they're defined.